guides/ADDING_A_NEW_PRODUCT.md

```diff
@@ -0,0 +1,160 @@
+the integreatly-operator to always install a specific version of each product, it maintains it's own set of manifests 
+for each product. To do this:
+- Create a new directory in the `manifests` directory, named: `integreatly-<product-name>`
+- Create a `<product-name>.package.yaml` file, the channel defined in this file must be `rhmi`.
+- Create a directory for the release of the operator to be deployed `<product>-<version>`.
+- Copy all the manifest files for the operator into this directory.
+
+**Note:** The operator installs the operator and product in separate namespaces and leverages the operatorgroup
+functionality of OLM to do this. In order to be compatible with how OLM does this the new CSV must define the namespace
+watched by the operator like so:
+```asciidoc
+- name: <WATCH_NAMESPACE_ENV_VAR>
+  valueFrom:
+    fieldRef:
+      fieldPath: metadata.annotations['olm.targetNamespaces']
+``` 
+
+## Define Product Variables 
+Every product has a series of variables defined in `pkg/apis/integreatly/v1alpha1/rhmi_types.go`, which are used through
+out the code-base, you will also need to define these for the new product.
+- ProductName this must be DNS-valid (i.e. all lower-case, and only alphanumeric and dashes)
+- ProductVersion
+- OperatorVersion
+
+## Add Product to Applicable Installation Types.
+For a product to be installed as part of an installation type, the configuration of that installation type needs to be
+updated to include the new product. These installation types are defined in `pkg/controller/installation/types.go`, at
+the time of writing, there are 2 installation types defined in variables in this file:
+- managed defined in `allManagedStages`
+- workshop defined in `allWorkshopStages`
+
+### Deciding when to create a new stage
+New stages are not desirable, as the operator will not progress to the following phase, until everything in the current
+phase has reported that it has completed, so adding new stages can slow down installation time.
+
+The idea of a stage is to allow the operator to complete some action, that will be required in the following stage, if 
+product A is required by product B, product A should probably go in an earlier stage than product B. For example, RHSSO
+requires cloud resource, so the cloud resources operator had to go into it's own stage, prior to the stage that RHSSO is 
+in.
+
+In general, if the new product is another tool for developers to use, it probably belongs in the products stage. Where 
+you add it inside a stage is immaterial; as these are not processed in any particular order, and nothing in a stage 
+should depend on something else in the same stage.
+
+## New Product Reconciler
+The reconciler must implement the `Products.Interface` interface, in order to work with  the installation controller. 
+The methods are defined in more detail below:
+
+### Reconcile
```

Comment:
It would be good to explain when, or how often, is the reconcile called.

Reply:
added a paragraph on this.